metadata/src/main/java/org/apache/kafka/controller/stream/StreamControlManager.java

```diff
@@ -625,14 +615,16 @@ public ControllerResult<CommitStreamSetObjectResponseData> commitStreamSetObject
                     ControllerResult<Errors> streamObjectCommitResult = this.s3ObjectControlManager.commitObject(streamObject.objectId(),
```

Comment:
Case exists: objectId=-1L, at this point streamObject should be used to determine idempotency.